rustica-agent/src/lib.rs

```diff
@@ -333,13 +334,27 @@ impl SshAgentHandler for Handler {
 
             return Ok(Response::SignResponse { signature });
         } else if let Signatory::Direct(privkey) = &self.signatory {
+            // Extract the pubkey fingerprint from either the SSH pubkey or the SSH cert
```

Comment:
Pull this out of the check and add it to the top of the function. Then we only need to do it once, and when we come back and fix this whole large if statement later it will be easier

Reply:
Fixed!